settlement-pipelines/src/bin/claim_settlement.rs

```diff
@@ -99,491 +118,370 @@ async fn main() -> anyhow::Result<()> {
+                resolve_combined_optional(data.merkle_tree_collection, data.settlement_collection)?,
+            ))
+        })
+        .collect::<anyhow::Result<HashMap<_, _>>>()?;
+    info!(
+        "Loaded json data from {:?} for epochs: {:?}",
+        settlement_json_files
+            .iter()
+            .map(|p| p.to_str())
+            .collect::<Vec<_>>(),
+        claiming_data.keys().collect::<Vec<_>>()
+    );
+    Ok(claiming_data)
 }
 
-fn process_merkle_trees_file(
+fn process_json_file(
     path: &PathBuf,
     loaded_data: &mut HashMap<u64, MerkleTreeLoadedData>,
 ) -> anyhow::Result<()> {
-    let path_string = path
+    let file_name = path
+        .file_name()
+        .ok_or(CliError::processing(format!(
+            "Cannot get file name from path: {:?}",
+            path
+        )))?
         .to_str()
-        .ok_or_else(|| anyhow!("Could not convert path {:?} to string", path))?;
+        .ok_or(CliError::processing(format!(
+            "Cannot convert file name to string: {:?}",
+            path
+        )))?;
     // Handle different file types based on suffix
-    if path_string.ends_with(SETTLEMENT_MERKLE_TREES_SUFFIX) {
+    if file_name.ends_with(SETTLEMENT_MERKLE_TREES_SUFFIX) {
         info!(
             "path: {:?} ends with {}",
             path, SETTLEMENT_MERKLE_TREES_SUFFIX
         );
-        let merkle_tree_collection: MerkleTreeCollection = read_from_json_file(path_string)
-            .map_err(|e| {
-                anyhow!(
+        let merkle_tree_collection: MerkleTreeCollection =
+            read_from_json_file(path).map_err(|e| {
+                CliError::processing(format!(
                     "Cannot read merkle tree collection from file '{:?}': {:?}",
-                    path,
-                    e
-                )
+                    path, e
+                ))
             })?;
         insert_merkle_tree_loaded_data(loaded_data, Some(merkle_tree_collection), None)?;
-    } else if path_string.ends_with(SETTLEMENTS_SUFFIX) {
+    } else if file_name.ends_with(SETTLEMENTS_SUFFIX) {
         info!("path: {:?} ends with {}", path, SETTLEMENTS_SUFFIX);
-        let settlement_collection: SettlementCollection = read_from_json_file(path_string)
-            .map_err(|e| {
-                anyhow!(
-                    "Cannot read settlement collection from from file '{:?}': {:?}",
-                    path,
-                    e
-                )
+        let settlement_collection: SettlementCollection =
+            read_from_json_file(path).map_err(|e| {
+                CliError::processing(format!(
+                    "Cannot read settlement collection from file '{:?}': {:?}",
+                    path, e
+                ))
             })?;
         insert_merkle_tree_loaded_data(loaded_data, None, Some(settlement_collection))?;
     }
 
     Ok(())
 }
 
+#[allow(clippy::too_many_arguments)]
+async fn claim_settlement<'a>(
+    program: &Program<Rc<DynSigner>>,
```

Comment:
RC again - can you please try to find all occurrences in the code?

Reply:
I know about it and will do the fix after this get processed. I created a separate task to fix this issue.
My plan is to finish the pipelines refactor and reporting fixes (this PR), then to develop the bonds config update and then to fix the issue with usage of Rc builder and pipeline code.